lib/matplotlib/artist.py

```diff
@@ -20,6 +20,30 @@
 
 _log = logging.getLogger(__name__)
 
+def _prevent_rasterization(draw):
+    # we assume that by default artists are not allowed to rasterize (unless
+    # its draw method is explicitly decorated). If it is being drawn after a
+    # rasterized artist and it has reached the rater_depth of 0. We stop
+    # rasterization so that it does not affect the behavior of normal artist
+    # (e.g., change in dpi). If the artist's draw method is decorated
+    # (draw._supports_rasterization is True), it won't  be decorated by
+    # `_prevent_rasterization`.
+
+    if hasattr(draw, "_supports_rasterization"):
```

Comment:
Do we also want to set `draw._supports_rasterization = False` in this wrapper?

Reply:
I have no preference, as far as we use the current condition of `getattr(self.draw, "_supports_rasterization", False)` .
And not sure which is better.

Reply:
On one hand, not setting it is the safer minimal change, but setting it to False (and finding + fixing our internal references to use `getattr(obj, '_supports_rasterization', False)` instead) makes everything more consistent and understandable for the next set of people.

Reply:
I will not block merging this over sorting that out.

Reply:
The only occurrence of `_supports_rasterization` is in the `Artist.set_rasterized` method (which already does ` getattr(obj, '_supports_rasterization', False)`. And I will go ahead and add `draw._supports_rasterization = False` here.